Index: avian_0.6+20121011/makefile
===================================================================
--- avian_0.6+20121011.orig/makefile	2012-10-11 10:38:25.496884663 +0200
+++ avian_0.6+20121011/makefile	2012-10-11 10:38:38.364070502 +0200
@@ -227,6 +227,9 @@
 soname-flag = -Wl,-soname -Wl,$(so-prefix)jvm$(so-suffix)
 version-script-flag = -Wl,--version-script=openjdk.ld
 
+cflags += $(DPKG_CFLAGS)
+lflags += $(DPKG_LFLAGS)
+
 build-system = posix
 
 system = posix
